src/Content/Text/BBCode.php

```diff
@@ -1903,6 +1896,32 @@ function ($matches) {
 		return $text;
 	}
 
+	/**
+	 * Get IFrame filter text
+	 *
+	 * @return string IFrame filter text
+	 */
+	private static function getIFrameFilter()
+	{
+		$filter = '%^(?:https://www.youtube.com/embed/|https://player.vimeo.com/video/';
+	
+		if (!DI::config()->get('system', 'no_oembed_rich_content')) {
+			$filter .= '|https:// ';
+		}
+		
+		$str_allowed = DI::config()->get('system', 'allowed_oembed', '');
+		if (!empty($str_allowed)) {
+			$allowed = explode(',', $str_allowed);
+			foreach ($allowed as $site) {
+				$filter .= '|https://.*' . trim($site);
```

Comment:
This would allow any arbitrary domain ending in one of the domains we allow, making the allowlist all but useless. We need to match on the exact domain. I have a similar patch for it I can submit. I wasn't able to formally test it given my troubles specified in https://github.com/friendica/friendica/issues/10019#issuecomment-798766397

Reply:
The problem with matching the exact host name is that for example for soundcloud the host to query is `soundcloud.com` but the iframe source is `m.soundcloud.com`.

Reply:
It definitely can, we can match either the full domain or sub-domains but this isn't what you did :D

Reply:
Fact is: We will have something like `soundcloud.com` in the list of accepted OEmbed sources - and we will have something like `https://m.soundcloud.com/` as iframe source. So we need a working regular expression that doesn't accept too much.

Reply:
I've had a similar approach to your, prepend the https scheme to the domains allowed in the list. It isn't a big step to add a condition on sub-domains. I have something that's almost ready, but now it's too late to finish and submit it.

Reply:
I think that we can assume `https`. So making this regular expression as safe as possible should be fine. I did some tests with my construction. But I really don't trust myself with regular expressions.

I'm looking forward to your solution!